src/sass/patterns/_buttons.scss

```
@@ -47,3 +49,23 @@ button {
     transform: scale(1.15);
   }
 }
+
+[data-btn~='switch'] {
+  --icon-size: var(--icon-medium);
+
+  block-size: fit-content;
+  margin-block-end: var(--spacer);
+
+  @include config.below('sm-page-break') {
+    [data-icon] {
+      transform: rotate(90deg);
+    }
+  }
+
+  &:hover,
+  &:focus {
+    --outline-width: thin;
+
+    animation: swap var(--fast) var(--out-back);
```

The `transform` property already has a transition on it, so might be simpler to rotate rather than add a full animation:

```suggestion
    transform: rotate(180deg);
```

---

removed the full animation in https://github.com/oddbird/oddcontrast/pull/227/commits/ad9a6f91d37930c236895a07aa48886120079d6f